tex/homology/cup-product.tex

```diff
@@ -577,11 +623,258 @@ \section{Wedge sums}
+
+And so, let us restate \Cref{thm:topological_kunneth_1}:
+\begin{theorem}
+	If $X$ and $Y$ are CW complexes, then
+	\[ \times \colon H_m(X) \otimes_\ZZ H_n(Y) \to H_{m + n}(X \times Y) \]
+	is an injective $\ZZ$-module homomorphism.
+\end{theorem}
+
+\subsection{Cross product on cellular homology}
+
+The definition with singular homology is quite clumsy ---
+because we use simplices as the building blocks for the chains,
+the product of two simplices in $X$ and $Y$ becomes a huge collection of simplices in $X \times Y$.
+
+We will now redefine the cross product using cellular homology ---
+it can be safely skipped, since both definitions of the cross product gives identical result on the
+homology groups.
+
+If $X$ and $Y$ are CW complexes, we can do better.
+We see that $X \times Y$ has a natural CW complex structure: for each cell $e^m$ of $X$ and cell
+$e^n$ of $Y$, their product makes for a cell $e^{m + n}$ of $X \times Y$.
+
+\begin{example}
+	If $X$ and $Y$ are both line segments built from
+	two $0$-cells and one $1$-cell, then their product $X \times Y$ has a natural CW complex
+	structure containing:
+	\begin{itemize}
+		\ii $4$ $0$-cells,
+		\ii $4$ $1$-cells,
+		\ii $1$ $2$-cell.
+	\end{itemize}
+\end{example}
+
+Recall the cellular groups $\Cells_\bullet(X)$ from \Cref{ch:cellular_homology}, each basis element
+corresponds to a cell in $X$.
+Then, we can define the cross product on the basis elements:
+\[ \times \colon \Cells_m(X) \otimes_\ZZ \Cells_n(Y) \to \Cells_{m + n}(X \times Y). \]
+To be painfully explicit:
+let $e^m \in \Cells_m(X)$, $e^n \in \Cells_m(Y)$, then the cross product is defined by $e^m \times
+e^n = e^m \times e^n \in \Cells_{m + n}(X \times Y)$ --- even the notation used is trivial.
+
+Of course, this induces a map on the homology groups:
+\[ \times \colon H_m(X) \otimes_\ZZ H_n(Y) \to H_{m + n}(X \times Y). \]
+This map is the same as the map we defined earlier.
+
+\subsection{Cross product on cellular cohomology}
+
+We do the same thing as above, but this time with cohomology --- remember that homology and
+cohomology are slightly different measures of ``holes'', for $K$ the Klein bottle then $H_2(X) =
+0$ but $H^2(X; \ZZ) \neq 0$.
+
+Given two cellular cochains $f \in \Hom(\Cells_m(X); R)$ and
+$g \in \Hom(\Cells_n(Y); R)$, we want to obtain a cochain $f
+\times g \in \Hom(\Cells_{m + n}(X \times Y); R)$.
+
+Of course, it is defined in the most natural way possible: for a cell $e^m$ of $X$ and a cell $e^n$
+of $Y$, we have $(f \times g)(e^m \times e^n) = f(e^m) \cdot g(e^n)$.
+
+Sounds good? Not yet --- since not all $(m + n)$-cells $e^{m + n}$ of $X \times Y$
+is formed as a product of a $m$-cell in $X$ and a $n$-cell in $Y$.
+For those, we simply declare that $(f \times g)(e^{m + n}) = 0$.
+
+As usual, this map induces a $R$-module homomorphism on the cohomology groups:
+\[ \times \colon H^m(X; R) \otimes_R H^n(Y; R) \to H^{m + n}(X \times Y; R). \]
```

Comment:
Hope I explained it clearly enough so the reader can e.g. intuitively compute the cross product of e.g. the two generators of H^1(S_1).